lib/views/frontend/spree/checkout/payment/_stripe.html.erb

```diff
@@ -1,71 +1,61 @@
-      Spree.stripePaymentMethod.find('#card_number, #card_expiry, #card_code').prop("disabled", true);
-      Spree.stripePaymentMethod.find(".ccType").prop("disabled", false);
-      Spree.stripePaymentMethod.find(".ccType").val(mapCC(response.card.brand));
-      token = response['id'];
-      paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][gateway_payment_profile_id]' value='" + token + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][last_digits]' value='" + response.card.last4 + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][month]' value='" + response.card.exp_month + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][year]' value='" + response.card.exp_year + "'/>");
-      return Spree.stripePaymentMethod.parents("form").get(0).submit();
-    }
+  function stripeTokenHandler(token) {
+    // Insert the token ID into the form so it gets submitted to the server
+    let form = $(Spree.stripePaymentMethod.parents("form").get(0));
+    let paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
+
+    addStripeHiddenInput(form, paymentMethodId, "gateway_payment_profile_id",  token.id);
+    addStripeHiddenInput(form, paymentMethodId, "last_digits",  token.card.last4);
+    addStripeHiddenInput(form, paymentMethodId, "month",  token.card.exp_month);
+    addStripeHiddenInput(form, paymentMethodId, "year",  token.card.exp_year);
+
+    form.submit();
   };
 
-  $(document).ready(function() {
-    Spree.stripePaymentMethod.prepend("<div id='stripeError' class='errorExplanation' style='display:none'></div>");
-    return $('#checkout_form_payment [data-hook=buttons]').click(function() {
-      var expiration, params;
-      $('#stripeError').hide();
-      if (Spree.stripePaymentMethod.is(':visible')) {
-        expiration = $('.cardExpiry:visible').payment('cardExpiryVal');
-        params = $.extend({
-          number: $('.cardNumber:visible').val(),
-          cvc: $('.cardCode:visible').val(),
-          exp_month: expiration.month || 0,
-          exp_year: expiration.year || 0
-        }, Spree.stripeAdditionalInfo);
-        Stripe.card.createToken(params, stripeResponseHandler);
-        return false;
+  function createToken() {
+    stripe.createToken(card).then((result) => {
+      if (result.error) {
+        // Inform the user if there was an error
+        var errorElement = document.getElementById('card-errors');
+        errorElement.textContent = result.error.message;
+      } else {
+        // Send the token to your server
+        stripeTokenHandler(result.token);
       }
     });
+  };
+
+  $(() => {
```

Comment:
Move all the functions into this scope.  Right now, having functions at the `<script>` level makes them global on the window object.  In practice, we should avoid this as they can conflict with other libraries/code.